src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

```diff
@@ -17,120 +22,168 @@
+                .build());
+
+        try {
+            filmDbStorage.removeLike(filmId, userId);
+        } catch (Exception e) {
+            log.info("User {} didn't like film {}. Skipping database operation.", userId, filmId);
         }
+    }
 
-        filmStorage.removeLike(filmId, userId);
+    public List<Film> getPopularFilms(int count, Integer genreId, Integer year) {
+        return filmDbStorage.getPopularFilms(count, genreId, year);
     }
 
-    public List<Film> getPopularFilms(int count) {
-        return filmStorage.getPopularFilms(count);
+    public List<Film> getCommonFilms(int userId, int friendId) {
+        return filmDbStorage.getCommonFilms(userId, friendId);
+    }
+
+    public List<Film> getRecommendedFilms(int id) {
+        return filmDbStorage.getRecommendedFilms(id);
     }
 
     public void addGenre(int filmId, Genre genre) {
-        Film film = filmStorage.getById(filmId);
+        log.info("Adding genre {} to film {}", genre.getId(), filmId);
+        Film film = filmDbStorage.getById(filmId);
         film.getGenres().add(genre);
-        filmStorage.update(film);
+        filmDbStorage.update(film);
     }
 
     public void removeGenre(int filmId, Genre genre) {
-        Film film = filmStorage.getById(filmId);
+        log.info("Removing genre {} from film {}", genre.getId(), filmId);
+        Film film = filmDbStorage.getById(filmId);
         film.getGenres().remove(genre);
-        filmStorage.update(film);
+        filmDbStorage.update(film);
     }
 
     private void validateFilmData(Film film) {
-        if (film.getMpa() == null) {
-            throw new ValidationException("Рейтинг MPA обязателен");
-        }
-
-        int mpaId = film.getMpa().getId();
-        validateMpa(mpaId);
+        if (film.getMpa() == null) throw new ValidationException("Рейтинг MPA обязателен");
+        validateMpa(film.getMpa().getId());
 
         Set<Integer> genreIds = extractGenreIds(film);
         validateGenres(genreIds);
     }
 
     private void validateMpa(int mpaId) {
-        if (mpaId < 1 || mpaId > 5) {
-            throw new NotFoundException("Рейтинг MPA с ID " + mpaId + " не найден");
-        }
+        if (mpaId < 1 || mpaId > 5) throw new NotFoundException("Рейтинг MPA с ID " + mpaId + " не найден");
     }
 
     private void validateGenres(Set<Integer> genreIds) {
         if (genreIds != null) {
             for (Integer genreId : genreIds) {
-                if (genreId < 1 || genreId > 6) {
+                if (genreId < 1 || genreId > 6)
                     throw new NotFoundException("Жанр с ID " + genreId + " не найден");
-                }
             }
         }
     }
 
     private Set<Integer> extractGenreIds(Film film) {
-        if (film.getGenres() == null) {
-            return new HashSet<>();
+        if (film.getGenres() == null) return new HashSet<>();
+        return film.getGenres().stream().map(Genre::getId).collect(Collectors.toSet());
```

Comment:
Стримы принято бить по строкам для лучшей читаемости кода:

```suggestion
        return film.getGenres().stream()
            .map(Genre::getId)
            .collect(Collectors.toSet());
```